server/src/main/java/org/opensearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -309,15 +310,15 @@ void buildDiffAndSerializeStates() {
                 try {
                     if (sendFullVersion || previousState.nodes().nodeExists(node) == false) {
                         if (serializedStates.containsKey(node.getVersion()) == false) {
-                            serializedStates.put(node.getVersion(), serializeFullClusterState(newState, node.getVersion()));
+                            serializedStates.put(node.getVersion(), CompressionHelper.serializedWrite(newState, node.getVersion(), true));
```

Comment:
you will clean up the ```serializeFullClusterState``` and ```serializeDiffClusterState``` from this class

Reply:
Yes, this is just draft PR.